pytorch_lightning/trainer/trainer.py

```diff
@@ -2616,6 +2617,41 @@ def weights_summary(self, val: Optional[str]) -> None:
     Other
     """
 
+    @property
+    def estimated_num_optimization_steps(self) -> int:
+        """Total optimization steps inferred from dataloaders, gradient accumulation factor and distributed
+        setup."""
+        accumulation_scheduler = self.accumulation_scheduler
+
+        if accumulation_scheduler.epochs != [0]:
+            raise MisconfigurationException(
+                "Estimated optimization steps cannot be computed with different"
+                " `accumulate_grad_batches` at different epochs."
+            )
+
+        # infinite training
+        if self.max_epochs == -1 and self.max_steps == -1:
+            return float("inf")
+
+        if self.train_dataloader is None:
+            rank_zero_warn("Loading `train_dataloader` to estimate number of optimization steps.")
```

Comment:
Why a warning?

```suggestion
            rank_zero_info("Loading `train_dataloader` to estimate number of optimization steps.")
```

Reply:
yeah agreed, it's nothing that the user can do to prevent this, so info is more appropriate imo.

Reply:
we wouldn't see an info message multiple times, right? only one time at the beginning of the training if we call it too early?

Reply:
yes..